Поддержка KSP2

Привет!

Пытался обновить проект на KSP 2.1.20-2.0.0 и от Yatagan прилетела ошибка при сборке. Как это исправить?

```
Caused by: java.lang.AssertionError: Unexpected retention
        at com.yandex.yatagan.lang.ksp.KspAnnotationImpl$AnnotationClassImpl.getRetention(KspAnnotationImpl.kt:199)
        at com.yandex.yatagan.core.graph.impl.RetentionKt.validateAnnotationsRetention(retention.kt:56)
        at com.yandex.yatagan.core.graph.impl.BindingGraphImpl.validate(BindingGraphImpl.kt:382)
        at com.yandex.yatagan.validation.impl.ValidatorImpl.inline(validation.kt:61)
        at com.yandex.yatagan.validation.impl.ValidationKt.validate(validation.kt:76)
        at com.yandex.yatagan.processor.common.ProcessKt.process(process.kt:60)
        at com.yandex.yatagan.processor.ksp.KspYataganProcessor.process(KspYataganProcessor.kt:47)
        at com.google.devtools.ksp.impl.KotlinSymbolProcessing$execute$1$1.invoke(KotlinSymbolProcessing.kt:579)
        at com.google.devtools.ksp.impl.KotlinSymbolProcessing$execute$1$1.invoke(KotlinSymbolProcessing.kt:577)
        at ksp.com.google.devtools.ksp.common.IncrementalContextBase.closeFilesOnException(IncrementalContextBase.kt:400)
        at com.google.devtools.ksp.impl.KotlinSymbolProcessing.execute(KotlinSymbolProcessing.kt:577)
        at com.google.devtools.ksp.impl.KSPLoader$Companion.loadAndRunKSP(KSPLoader.kt:37)
        at com.google.devtools.ksp.impl.KSPLoader.loadAndRunKSP(KSPLoader.kt)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source)
        at java.base/java.lang.reflect.Method.invoke(Unknown Source)
        at com.google.devtools.ksp.gradle.KspAAWorkerAction.execute(KspAATask.kt:580)
```

--- a/lang/ksp/src/main/kotlin/com/yandex/yatagan/lang/ksp/KspAnnotationImpl.kt
+++ b/lang/ksp/src/main/kotlin/com/yandex/yatagan/lang/ksp/KspAnnotationImpl.kt
@@ -116,6 +116,18 @@ internal class KspAnnotationImpl(
                         visitor.visitType(KspTypeImpl(impl = value))
                     }
                 }
+                is KSClassDeclaration -> {
+                    val declaration = value
+                    if (declaration.classKind == ClassKind.ENUM_ENTRY) {
+                        val enumDeclaration = declaration.parentDeclaration as KSClassDeclaration
+                        visitor.visitEnumConstant(
+                            enum = KspTypeImpl(impl = enumDeclaration.asType(emptyList())),
+                            constant = declaration.simpleName.asString(),
+                        )
+                    } else {
+                        visitor.visitType(KspTypeImpl(value.asStarProjectedType()))
+                    }
+                }
                 is KSAnnotation -> visitor.visitAnnotation(KspAnnotationImpl(this@KspAnnotationImpl, value))
                 is List<*> -> visitor.visitArray(value.map { ValueImpl(it ?: "<error>") })
                 is Enum<*> -> {
@@ -204,6 +216,12 @@ internal class KspAnnotationImpl(
                             "RUNTIME" -> AnnotationRetention.RUNTIME
                             else -> throw AssertionError("Unexpected retention")
                         }
+                        is KSClassDeclaration -> when (value.simpleName.getShortName()) {
+                            "SOURCE" -> AnnotationRetention.SOURCE
+                            "BINARY" -> AnnotationRetention.BINARY
+                            "RUNTIME" -> AnnotationRetention.RUNTIME
+                            else -> throw AssertionError("Unexpected retention")
+                        }
                         else -> throw AssertionError("Unexpected retention")
                     }
                     Utils.javaRetentionClass -> when (val value = annotation["value"]) {
@@ -215,6 +233,12 @@ internal class KspAnnotationImpl(
                         is KSType -> when (value.declaration.simpleName.getShortName()) {
                             "SOURCE" -> AnnotationRetention.SOURCE
                             "CLASS" -> AnnotationRetention.BINARY
+                            "RUNTIME" -> AnnotationRetention.RUNTIME
+                            else -> throw AssertionError("Unexpected retention")
+                        }
+                        is KSClassDeclaration -> when (value.simpleName.getShortName()) {
+                            "SOURCE" -> AnnotationRetention.SOURCE
+                            "BINARY" -> AnnotationRetention.BINARY
                             "RUNTIME" -> AnnotationRetention.RUNTIME
                             else -> throw AssertionError("Unexpected retention")
                         }

